### ISSUE TITLE
Add support for Request Retry

### ISSUE THREAD

**Issue by jeevatkm** [go-resty/resty#24]
It would be nice to have request try with interval, no of attempts, etc.

**Comment by dahankzter** [comment]
Very nice yes! And with exponential back off as well if possible.

**Comment by jeevatkm** [comment]
Not much of pressure here, I'm taking it very slow on this :-)

**Comment by dahankzter** [comment]
It would be very useful though ;)

**Comment by alaska** [comment]
I could take this on. I'm writing a little retry library myself.

**Comment by jeevatkm** [comment]
@alaska please let me know once you're library is ready, I will have a look.

**Comment by jeevatkm** [comment]
@keithballdotnet Contributed Backoff retry #34

**Comment by jeevatkm** [comment]
#35 is merged.
